met/src/tools/core/point_stat/point_stat.cc

```diff
@@ -658,138 +661,68 @@ void process_obs_file(int i_nc) {
    // Open the observation file as a NetCDF file.
    // The observation file must be in NetCDF format as the
    // output of the PB2NC or ASCII2NC tool.
-   obs_in = open_ncfile(obs_file[i_nc].c_str());
-
-   if(IS_INVALID_NC_P(obs_in)) {
-      delete obs_in;
-      obs_in = (NcFile *) 0;
+   MetNcPointObsIn nc_point_obs;
+   if( !nc_point_obs.open(obs_file[i_nc].c_str()) ) {
+      nc_point_obs.close();
 
-      mlog << Warning << "\nprocess_obs_file() -> "
+      mlog << Warning << "\n" << method_name
            << "can't open observation netCDF file: "
            << obs_file[i_nc] << "\n\n";
       return;
    }
 
-   // Read the dimensions and variables
-   NetcdfObsVars obs_vars;
-   read_nc_dims_vars(obs_vars, obs_in);
-
-   bool use_var_id = obs_vars.use_var_id;
-   if (use_var_id) {
-      NcDim var_dim = get_nc_dim(obs_in,nc_dim_nvar);
-      get_dim_size(&var_dim);
-   }
-
-   int exit_code = check_nc_dims_vars(obs_vars);
-   if(exit_code == exit_code_no_dim) {
-      mlog << Error << "\nprocess_obs_file() -> "
-           << "can't read \"mxstr\", \"nobs\" or \"nmsg\" "
-           << "dimensions from netCDF file: "
-           << obs_file[i_nc] << "\n\n";
-      exit(1);
-   }
-
-   if(exit_code == exit_code_no_hdr_vars) {
-      mlog << Error << "\nprocess_obs_file() -> "
-           << "can't read \"hdr_typ\", \"hdr_sid\", "
-           << "or \"hdr_vld\" variables from netCDF file: "
-           << obs_file[i_nc] << "\n\n";
-      exit(1);
-   }
-
-   if(exit_code == exit_code_no_loc_vars) {
-      mlog << Error << "\nprocess_obs_file() -> "
-           << "can't read \"hdr_arr\" or \"hdr_lat\" "
-           << "variables from netCDF file: "
-           << obs_file[i_nc] << "\n\n";
-      exit(1);
-   }
-   if(exit_code == exit_code_no_obs_vars) {
-      mlog << Error << "\nprocess_obs_file() -> "
-           << "can't read \"obs_arr\" or \"obs_val\" "
-           << "variables from netCDF file: "
-           << obs_file[i_nc] << "\n\n";
-      exit(1);
-   }
-
-   if(IS_INVALID_NC(obs_vars.obs_qty_var))
-      mlog << Debug(3) << "Quality marker information not found input file\n";
-
-   int obs_count = get_dim_size(&obs_vars.obs_dim);
-   int hdr_count = get_dim_size(&obs_vars.hdr_dim);
+   nc_point_obs.read_dim_headers();
+   nc_point_obs.check_nc(obs_file[i_nc].c_str(), method_name);
+   nc_point_obs.read_obs_data_strings();
```

Comment:
Could change to a better function name including "table lookup" but its up to you.